src/layers/data_type_layer.cpp

```diff
@@ -302,10 +305,10 @@ auto data_type_layer<TensorDataType>::get_activations(const Layer& child) const
 }
 template <typename TensorDataType>
 auto data_type_layer<TensorDataType>::get_error_signals(const Layer& parent) const -> const BaseDistMat& {
-  const int parent_index = (std::find(m_parent_layers.begin(),
-                                      m_parent_layers.end(),
-                                      &parent)
-                           - m_parent_layers.begin());
+  const int parent_index = std::distance(m_parent_layers.begin(),
```

Comment:
If we have a method for computing the child's index, we should have the same for the parent's index.